parsers/src/main/java/org/semanticweb/owlapi/rdf/syntax/RDFParser.java

```diff
@@ -84,29 +84,45 @@
 @SuppressWarnings("unused")
 public class RDFParser extends DefaultHandler implements RDFConstants {
 
-    protected static final Locator s_nullDocumentLocator = new LocatorImpl();
-    protected static final SAXParserFactory s_parserFactory = buildFactory();
+    private static ThreadLocal<Locator> locatorThreadLocal = new ThreadLocal<Locator>() {
+        @Override
+        protected Locator initialValue() {
+            return new LocatorImpl();
+        }
+    };
+
+    protected static Locator getLocator() {
+        return locatorThreadLocal.get();
+    }
+    private static ThreadLocal<SAXParserFactory> saxParserFactoryThreadLocal = new ThreadLocal<SAXParserFactory>() {
+
+        @Override
+        protected SAXParserFactory initialValue() {
+            try {
+                System.setProperty("entityExpansionLimit", "100000000");
```

Comment:
We had troubles with threadlocals in Tomcat a long time back - if I remember correctly, around the caches for IRI prefixes and other similar caches. I don't think the creation of a SaxParserFactory happens often enough or is expensive enough to bother caching it - we could do this the same way it's done in version 5

https://github.com/owlcs/owlapi/blob/version5/api/src/main/java/org/semanticweb/owlapi/util/SAXParsers.java

It's not perfect because RDFParser still has a static reference, so that needs addressing as well.
Good work spotting this, by the way.